deviceaccess.py

```diff
@@ -1381,3 +1426,13 @@ def write(
         accessModeFlags = [] if accessModeFlags is None else accessModeFlags
         self._device.write(array, registerPath, numberOfElements,
                            wordOffsetInRegister, accessModeFlags)
+
+#######################################################################################################################
+# Type Definitions
+
+
+UserType = reduce(lambda x, y: Union[x, y], list(Device._userTypeExtensions.values()))
+
+RegisterAccessor = Union[OneDRegisterAccessor,
+                         TwoDRegisterAccessor,
+                         ScalarRegisterAccessor]
```

Comment:
Shouldn't these be placed above their usage?

Reply:
Device is not known at the start of the file, so it has to be below Device.

Annotations can be defined after they are used, as they are evaluated later. https://docs.python.org/3/howto/annotations.html

Reply:
all right, thanks for the explanation.